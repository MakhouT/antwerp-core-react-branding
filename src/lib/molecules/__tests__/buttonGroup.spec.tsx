import * as React from "react";
import * as renderer from "react-test-renderer";

import * as A from "../..";

test("ButtonGroup", () => {
    // Act
    const component: any = renderer.create(
        <A.ButtonGroup>
            <A.Button>bla</A.Button>
            <A.Button>bli</A.Button>
        </A.ButtonGroup>);
    const tree: renderer.ReactTestRendererJSON = component.toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
});

test("ButtonGroup without children", () => {
    // Act
    const component: any = renderer.create(
        <A.ButtonGroup>
        </A.ButtonGroup>);
    const tree: renderer.ReactTestRendererJSON = component.toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
});

test("ButtonGroup with wrong children", () => {
    // Act
    const component: any = renderer.create(
        <A.ButtonGroup>
            <A.Button>bla</A.Button>
            <div>Wrong child</div>
        </A.ButtonGroup>);
    const tree: renderer.ReactTestRendererJSON = component.toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
});
