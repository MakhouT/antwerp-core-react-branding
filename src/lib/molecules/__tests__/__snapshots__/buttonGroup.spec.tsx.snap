// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup 1`] = `
<div
  className="m-button-group"
>
  <button
    autoFocus={true}
    className="a-button"
    disabled={undefined}
    onClick={[Function]}
    style={undefined}
    tabIndex={undefined}
    type="button"
  >
    bla
  </button>
  <button
    autoFocus={true}
    className="a-button"
    disabled={undefined}
    onClick={[Function]}
    style={undefined}
    tabIndex={undefined}
    type="button"
  >
    bli
  </button>
</div>
`;

exports[`ButtonGroup with wrong children 1`] = `
<div
  className="m-button-group"
>
  <button
    autoFocus={true}
    className="a-button"
    disabled={undefined}
    onClick={[Function]}
    style={undefined}
    tabIndex={undefined}
    type="button"
  >
    bla
  </button>
</div>
`;

exports[`ButtonGroup without children 1`] = `
<div
  className="m-button-group"
/>
`;
